### PR TITLE
 [Python] Custom Teams Endpoint (CTE) GA

### PR DESCRIPTION
**Pull request by AikoBB**
# Description

Preparing CTE for the GA:

- Added mandatory parameters(appId, userId) to `getTokenForTeamsUser`
- Added GA API version to `ServiceVersion`

## Implementation TODOs
- [x] Update Identity SDK to point to new CTE path
  - [x] regenerate implementation from a new swagger spec
  - [x] update recordings
- [x] Remove all occurrences of api_version parameter in other SDKs within the initialization of CommunicationIdentityClient
- [ ] changes regarding to ACS verification of App ID & UserId
  - [x] regenerate implementation from a new swagger spec - https://github.com/Azure/azure-rest-api-specs/pull/17930
  - [x] update client implementation
  - [x] update & add tests
  - [x] update recordings
  - [x] update README & CHANGELOG
  - [x] update tag and require path in swagger/README.md
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by petrsvihlik** [comment]
@AikoBB if you by any chance read this, could you please give me contributor rights to your branch?

File: sdk/communication/azure-communication-identity/tests/recordings/test_communication_identity_client.test_get_token_for_teams_user_with_invalid_token.yaml
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/tests/recordings/test_communication_identity_client.test_get_token_for_teams_user_with_invalid_token.yaml:22]: `getToken` Seems that we need to regenerate the rest client using autorest and re-record the mocks

Comment (petrsvihlik) [reply sdk/communication/azure-communication-identity/tests/recordings/test_communication_identity_client.test_get_token_for_teams_user_with_invalid_token.yaml:22]: or maybe this is a leftover file that can be deleted?

File: sdk/communication/azure-communication-identity/azure/communication/identity/aio/_communication_identity_client_async.py
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/azure/communication/identity/aio/_communication_identity_client_async.py:194]:
```suggestion
        :param aad_token: an Azure AD access token of a Teams User
```

File: sdk/communication/azure-communication-identity/dev_requirements.txt
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/dev_requirements.txt:7]: not sure what is this. can you pls verify?

File: sdk/communication/azure-communication-identity/azure/communication/identity/_generated/aio/operations/_communication_identity_operations.py
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/azure/communication/identity/_generated/aio/operations/_communication_identity_operations.py:179]: This is quite interesting...that the generator has removed the api_version kwarg everywhere. You know why is that? Is it because we're targeting a stable version again?

File: sdk/communication/azure-communication-identity/CHANGELOG.md
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/CHANGELOG.md:6]: The same comments I had for other platforms apply here.

File: sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py:199]:
```suggestion
        :param aad_token: an Azure AD access token of a Teams User.
```

File: sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py:193]: Just a note: I think we want to keep these expanded and not wrapped in an options bag as on other platforms. It's not idiomatic for Python to have options bags.

File: sdk/communication/azure-communication-identity/tests/test_communication_identity_client.py
Comment (petrsvihlik) [inline sdk/communication/azure-communication-identity/tests/test_communication_identity_client.py:334]: the same as for other languages - I wonder if this was supposed to change.

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-identity/tests/test_communication_identity_client.py
Comment (AikoBB) [reply sdk/communication/azure-communication-identity/tests/test_communication_identity_client.py:334]: we have already answered [here](https://github.com/Azure/azure-sdk-for-net/pull/26924#discussion_r878068558)

File: sdk/communication/azure-communication-identity/dev_requirements.txt
Comment (AikoBB) [reply sdk/communication/azure-communication-identity/dev_requirements.txt:7]: this library for testing which provides features running tests with different value of the parameters

File: sdk/communication/azure-communication-identity/tests/recordings/test_communication_identity_client.test_get_token_for_teams_user_with_invalid_token.yaml
Comment (AikoBB) [reply sdk/communication/azure-communication-identity/tests/recordings/test_communication_identity_client.test_get_token_for_teams_user_with_invalid_token.yaml]: yeah, they were leftovers, deleted

File: sdk/communication/azure-communication-identity/azure/communication/identity/_generated/aio/operations/_communication_identity_operations.py
Comment (AikoBB) [reply sdk/communication/azure-communication-identity/azure/communication/identity/_generated/aio/operations/_communication_identity_operations.py:179]: They were moved to the [clients of the public API layer](https://github.com/Azure/azure-sdk-for-python/blob/bb27e0594f1bf2c15dac655598cc615c3b194d8d/sdk/communication/azure-communication-identity/azure/communication/identity/_generated/_communication_identity_client.py#L35) which are the main interaction points for 
the convenience layer

Comment (petrsvihlik) [reply sdk/communication/azure-communication-identity/azure/communication/identity/_generated/aio/operations/_communication_identity_operations.py:179]: nice

**Review by petrsvihlik — APPROVED** [review]

**Comment by petrsvihlik** [comment]
Do not merge until we get the ARB signoff

File: sdk/communication/azure-communication-identity/CHANGELOG.md
Comment (annatisch) [inline sdk/communication/azure-communication-identity/CHANGELOG.md:3]: Why did we jump from 1.1.0b1 to 1.2.0? Shouldn't the next GA release be 1.1.0?

Comment (AikoBB) [reply sdk/communication/azure-communication-identity/CHANGELOG.md:3]: yes, it should be 1.1.0, I will fix it. Thanks!

Comment (AikoBB) [reply sdk/communication/azure-communication-identity/CHANGELOG.md:3]: This release will include dropping Python 2.x support which is a minor change([#20691](https://github.com/Azure/azure-sdk-for-python/issues/20691)), so that is why we jumped to 1.2.0

Comment (annatisch) [reply sdk/communication/azure-communication-identity/CHANGELOG.md:3]: 1.1.0 Never GA'ed - so you can roll the dropping of 2.7 into that release and it will still be part of a minor bump :)

**Pull request merged**
